cogs/reward.py

```diff
@@ -43,89 +45,118 @@ def convert_to_seconds(s):
+                        
 
-            if totalTime == 0:
-                await channel.send(content="You may have formatted the time incorrectly or calculated for 0. Try again with the correct format." + rewardCommand)
-                return
-            
-            t = 0
             if charDict["Level"] < 5:
-                t = 0
+                tierNum = 0
                 tierInitial = 'T1 TP'
                 tierNext = 'T2 TP'
+                tierNext1 = 'T3 TP'
+                tierNext2 = 'T4 TP'
+                tierNext3 = 'T5 TP'
             elif charDict["Level"] < 11:
-                t = 1
+                tierNum = 1
                 tierInitial = 'T2 TP'
                 tierNext = 'T3 TP'
+                tierNext1 = 'T4 TP'
+                tierNext2 = 'T5 TP'
             elif charDict["Level"] < 17:
-                t = 2
+                tierNum = 2
                 tierInitial = 'T3 TP'
                 tierNext = 'T4 TP'
+                tierNext1 = 'T5 TP'
             elif charDict["Level"] < 20:
-                t = 3
+                tierNum = 3
                 tierInitial = 'T4 TP'
                 tierNext = 'T5 TP'
-            elif charDict["Level"] == 20:
-                t = 4
+            else:
+                tierNum = 4
                 tierInitial = 'T5 TP'
             
-            treasureArray  = calculateTreasure(charDict["Level"], charDict["CP"] , t, totalTime)
+            # Uses calculateTreasure to determine the rewards from the quest based on the character
+            treasureArray  = calculateTreasure(charDict["Level"], charDict["CP"] , tierNum, totalTime)
             durationString = timeConversion(totalTime)
             treasureString = f"{treasureArray[0]} CP, {sum(treasureArray[1].values())} TP, {treasureArray[2]} GP"
             resultLevel = charDict["Level"]
             resultCP = charDict["CP"] + treasureArray[0]
-            resultLevel += resultCP // 10
-            nextCP = resultCP % 10
+            
+            if tierNum == 0 and resultCP > 4:   # For when levels are gained in tier 1
+                while resultLevel < 5:
+                    resultLevel += 1
+                    resultCP -= 4
+                    if resultCP < 5: # break the loop if they wouldn't level past 4
+                        nextCP = resultCP
+                        break
+                    if resultLevel == 5: #if they go outside of tier 1
+                        resultLevel += resultCP // 10
+                        nextCP = resultCP % 10
+            elif tierNum == 0:
+                nextCP = resultCP
+            else:
+                resultLevel += resultCP // 10
+                nextCP = resultCP % 10
+
+            if resultLevel > 20:    # If tier 5 is reached, level is capped at 20
+                levelDifference = resultLevel - 20
+                nextCP += (levelDifference * 10)
+                resultLevel = 20
```

Comment:
There is a far condensed version of this in char.py in the create function. You could extract it if you feel it is going to be reused more often. Right now the version I created is used only for create, respec and xfer iirc.

If you have an issue with my implementation please tell me